### PR TITLE
Chore/flatten project files to pkg

### PR DESCRIPTION
**Pull request by EasterTheBunny**

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,2 +1,2 @@
-golang 1.19
+golang 1.20
```
[the remark this answers is not in the document]
Comment (EasterTheBunny) [reply .tool-versions:1]: Not sure. I don't remember why we have tool versions. I'll add it to my list of things to review.

File: pkg/chain/evmreports_test.go [hunk 2]
```diff
@@ -168,8 +168,8 @@ func BenchmarkEncodeReport(b *testing.B) {
 	largeData := make([]byte, 128)
 	zeroBigInt := big.NewInt(0)
 
-	rand.Read(smallData)
-	rand.Read(largeData)
+	_, _ = rand.Read(smallData)
+	_, _ = rand.Read(largeData)
```
[the remark this answers is not in the document]
Comment (EasterTheBunny) [reply pkg/chain/evmreports_test.go:172]: Yup. Haha!

File: pkg/factory.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+		return nil, info, err
+	}
+
+	// for each of the provided dependencies, check if they satisfy a start/stop
+	// interface. if so, add them to a services array so that the plugin can
+	// shut them down.
+	possibleSrvs := []interface{}{coordinator, condObserver}
+	subProcs := make([]PluginCloser, 0, len(possibleSrvs))
+	for x := 0; x < len(possibleSrvs); x++ {
+		sub, ok := possibleSrvs[x].(PluginCloser)
```
[the remark this answers is not in the document]
Comment (EasterTheBunny) [reply pkg/factory.go:89]: Changed.

File: pkg/ocr.go [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+// Report implements the types.ReportingPlugin interface in OC2. This method
+// chooses a single upkeep from the provided observations by the earliest block
+// number, checks the upkeep, and builds a report. Multiple upkeeps in a single
+// report is supported by how the data is abi encoded, but no gas estimations
+// exist yet.
+func (p *ocrPlugin) Report(ctx context.Context, t types.ReportTimestamp, _ types.Query, attributed []types.AttributedObservation) (bool, types.Report, error) {
+	lCtx := newOcrLogContext(t)
+	ctx = context.WithValue(ctx, ocrLogContextKey{}, lCtx)
+
+	// --------- Condition Upkeep OCR Logic to Collect Keys from Observations -----------
```
[the remark this answers is not in the document]
Comment (EasterTheBunny) [reply pkg/ocr.go:160]: I agree. They are there for now to indicate that we COULD break them out. I was leaving them in place for now until the log trigger stuff was added as that could change things up a bit.

File: pkg/observation.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+	for i, obs := range attr {
+		// a single observation returning an error here can void all other
+		// good observations. ensure this loop continues on error, but collect
+		// them and throw an error if ALL observations fail at this point.
+		var upkeepObservation Observation
+		if err := decode(obs.Observation, &upkeepObservation); err != nil {
+			logger.Printf("unable to decode observation: %s", err.Error())
+			parseErrors++
+			continue
```
[the remark this answers is not in the document]
Comment (EasterTheBunny) [reply pkg/observation.go:86]: Correct. As long as the block key is there, we shouldn't care about the upkeep ids present.

**Review by ferglor — APPROVED** [review]

**Pull request merged**